File: backend/models/Staff.js
// backend/models/Staff.js
import mongoose from 'mongoose';
import bcrypt from 'bcryptjs';

const staffSchema = new mongoose.Schema({
  name: { type: String, required: true },
  email: { type: String, required: true, unique: true },
  password: { type: String, required: true, select: false },
  phone: String,
  facility: { type: mongoose.Schema.Types.ObjectId, ref: 'Facility' },
  role: { type: String, enum: ['staff', 'manager'], default: 'staff' },
  isActive: { type: Boolean, default: true },
  lastLogin: Date
}, { timestamps: true });

staffSchema.pre('save', async function(next) {
  if (!this.isModified('password')) return next();
  this.password = await bcrypt.hash(this.password, 12);
  next();
});

export default mongoose.model('Staff', staffSchema);

File: backend-go/internal/models/staff.go
package models

import (
	"go.mongodb.org/mongo-driver/bson/primitive"
	"time"
)

type Staff struct {
	ID        primitive.ObjectID  `bson:"_id,omitempty" json:"id,omitempty"`
	Name      string              `bson:"name" json:"name"`
	Email     string              `bson:"email" json:"email"`
	Password  string              `bson:"password" json:"-"` // not returned
	Phone     string              `bson:"phone,omitempty" json:"phone,omitempty"`
	Facility  *primitive.ObjectID `bson:"facility,omitempty" json:"facility,omitempty"`
	Role      string              `bson:"role" json:"role"` // staff, manager
	IsActive  bool                `bson:"isActive" json:"isActive"`
	LastLogin *time.Time          `bson:"lastLogin,omitempty" json:"lastLogin,omitempty"`
	CreatedAt time.Time           `bson:"createdAt" json:"createdAt"`
	UpdatedAt time.Time           `bson:"updatedAt" json:"updatedAt"`
}
